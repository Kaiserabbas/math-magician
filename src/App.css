/* Nav-Bar */

.nav-list {
  background-color: whitesmoke;
  display: flex;
  align-items: center;
  justify-content: right;
  list-style: none;
  gap: 20px;
  padding: 20px 50px 20px 0;
  font-family: poppins, sans-serif;
  font-size: 20px;
  color: black;
  margin: 40px 0 0 0;
  border-radius: 10px;
  box-shadow: 0 0 10px rgba(0, 0, 0, 0.2);
}

.title {
  position: absolute;
  left: 5%;
  font-size: 3rem;
}

.nav {
  text-decoration: none;
}

.nav:active {
  color: blue;
}

.nav::after {
  content: "|";
  margin: 0 10px;
}

.nav:last-child::after {
  content: none;
}

/* Nav-Bar */

/* Home Page */
.home {
  margin: 30px;
  font-family: poppins, sans-serif;
  margin-left: 5%;
}

.home h2 {
  font-size: 2.3rem;
}

/* Home Page */

/* Calculator */

.calculator-container {
  display: flex;
}

.calculator-text {
  margin-left: 5%;
  font-family: poppins, sans-serif;
}

.calculator-text h2 {
  font-size: 2.3rem;
}

.calculator {
  width: 500px;
  height: auto;
  background-color: #f4f4f4;
  border: 1px solid #ccc;
  border-radius: 8px;
  box-shadow: 0 0 10px rgba(0, 0, 0, 0.2);
  margin: 50px auto 0 auto;
  padding: 20px;
}

.display {
  height: 60px;
  background-color: #222;
  color: #fff;
  font-size: 2.5rem;
  text-align: right;
  padding: 10px;
  border-radius: 6px;
  margin-bottom: 10px;
}

.buttons {
  display: grid;
  grid-template-rows: repeat(4, 1fr);
  grid-gap: 10px;
}

.button {
  width: 100%;
  height: 65px;
  padding: 10px;
  border-radius: 6px;
  font-size: 20px;
  border: 1px solid #ccc;
  background-color: #e0e0e0;
  cursor: pointer;
  transition: background-color 0.3s;
}

.row {
  display: flex;
  gap: 10px;
}

.button:hover {
  background-color: #ccc;
}

.operator {
  background-color: #f49313dd;
  color: white;
  font-weight: bold;
}

.row5 {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
}

.row5 :first-child {
  grid-column: span 2;
}

/* Calculator */

/* Quote */

.quote-container {
  margin: 100px 0 0 0;
  font-family: poppins, sans-serif;
}

.quote {
  color: #000;
  position: relative;
  width: auto;
  font-style: italic;
  font-size: 2rem;
  text-align: center;
  margin: 50px 150px;
}

/* Quote */

@media screen and (max-width: 768px) {
  .title {
    position: relative;
  }

  .home {
    text-align: center;
  }

  .nav-list {
    flex-direction: column;
    justify-content: center;
    margin: 40px 0 0 0;
  }

  .calculator-container {
    flex-direction: column;
  }

  .calculator {
    width: 300px;
  }

  .calculator-text {
    text-align: center;
  }

  .display {
    height: 40px;
    font-size: 2rem;
  }

  .button {
    height: 45px;
  }

  .quote-container h1 {
    text-align: center;
  }

  .quote {
    margin: 50px;
  }
}
