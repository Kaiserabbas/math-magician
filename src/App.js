import React from 'react';
import { Routes, Route, Link } from 'react-router-dom';
import Calculator from './components/Calculator.js';
import QuoteDisplay from './components/QuoteDisplay.js';
import Home from './components/Home.js';
import './App.css';

function App() {
  return (
    <>
      <nav className="nav-list">
        <h1 className="title">Math Magician</h1>
        <Link className="nav" to="/" data-testid="home-component">
          Home
        </Link>
        <Link
          className="nav"
          to="/calculator"
          data-testid="calculator-component"
        >
          Calculator
        </Link>
        <Link
          className="nav"
          to="/quotes"
          data-testid="quote-display-component"
        >
          Quotes
        </Link>
      </nav>
      <Routes>
        <Route path="/" element={<Home />} />
        <Route path="/calculator" element={<Calculator />} />
        <Route path="/quotes" element={<QuoteDisplay />} />
      </Routes>
    </>
  );
}

export default App;
