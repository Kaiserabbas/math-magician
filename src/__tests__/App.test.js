import '@testing-library/jest-dom';
import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import App from '../App';

describe('App component', () => {
  it('renders navigation links correctly', () => {
    render(
      <MemoryRouter>
        <App />
      </MemoryRouter>
    );

    const homeLink = screen.getByTestId('home-component');
    const calculatorLink = screen.getByTestId('calculator-component');
    const quotesLink = screen.getByTestId('quote-display-component');

    expect(homeLink).toBeInTheDocument();
    expect(calculatorLink).toBeInTheDocument();
    expect(quotesLink).toBeInTheDocument();

    // Generate snapshot
    expect(screen.getByTestId('home-component')).toMatchSnapshot();
    expect(screen.getByTestId('calculator-component')).toMatchSnapshot();
    expect(screen.getByTestId('quote-display-component')).toMatchSnapshot();
  });

  it('renders the Home component when navigating to the root path', () => {
    render(
      <MemoryRouter initialEntries={['/']}>
        <App />
      </MemoryRouter>
    );

    const homeComponent = screen.getByTestId('home-component');

    expect(homeComponent).toBeInTheDocument();
  });

  it('renders the Calculator component when navigating to the calculator path', () => {
    render(
      <MemoryRouter initialEntries={['/calculator']}>
        <App />
      </MemoryRouter>
    );

    const calculatorComponent = screen.getByTestId('calculator-component');

    expect(calculatorComponent).toBeInTheDocument();
  });

  it('renders the QuoteDisplay component when navigating to the quotes path', () => {
    render(
      <MemoryRouter initialEntries={['/quotes']}>
        <App />
      </MemoryRouter>
    );

    const quoteDisplayComponent = screen.getByTestId('quote-display-component');

    expect(quoteDisplayComponent).toBeInTheDocument();
  });
});
