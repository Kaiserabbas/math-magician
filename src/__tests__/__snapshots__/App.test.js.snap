// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders navigation links correctly 1`] = `
<a
  class="nav"
  data-testid="home-component"
  href="/"
>
  Home
</a>
`;

exports[`App component renders navigation links correctly 2`] = `
<a
  class="nav"
  data-testid="calculator-component"
  href="/calculator"
>
  Calculator
</a>
`;

exports[`App component renders navigation links correctly 3`] = `
<a
  class="nav"
  data-testid="quote-display-component"
  href="/quotes"
>
  Quotes
</a>
`;
